File: galleries/tiny/seaborn_obj/grouped_bar.py
import seaborn.objects as so
import pandas as pd

df = pd.DataFrame(
    {
        "Fruit": ["Apples", "Oranges", "Bananas", "Apples", "Oranges", "Bananas"],
        "Contestant": ["Alex", "Alex", "Alex", "Jordan", "Jordan", "Jordan"],
        "Number Eaten": [2, 1, 3, 1, 3, 2],
    }
)

p = so.Plot(df, x="Fruit", y="Number Eaten", color="Contestant").add(
    so.Bar(), so.Dodge()
)

File: galleries/tiny/observable/grouped_bar.mjs
import * as Plot from "@observablehq/plot";

export async function chart() {
  const data = [
    { Fruit: "Apples", Contestant: "Alex", "Number Eaten": 2 },
    { Fruit: "Oranges", Contestant: "Alex", "Number Eaten": 1 },
    { Fruit: "Bananas", Contestant: "Alex", "Number Eaten": 3 },
    { Fruit: "Apples", Contestant: "Jordan", "Number Eaten": 1 },
    { Fruit: "Oranges", Contestant: "Jordan", "Number Eaten": 3 },
    { Fruit: "Bananas", Contestant: "Jordan", "Number Eaten": 2 },
  ];

  return Plot.auto(data, {
    y: "Number Eaten",
    x: "Contestant",
    fx: "Fruit",
    color: "Contestant",
    mark: "bar",
  }).plot({
    color: { legend: true },
  });
}
